conjur/ssl_service.py

```diff
@@ -37,7 +44,22 @@ def __connect(cls, hostname, port):
         """
         Method for opening a socket to the Conjur server
         """
-        ctx = SSL.Context(method=SSL.TLSv1_2_METHOD)
+        # Makes the TLS version configurable through an ENV variable
+        if os.getenv('TLS_VERSION') in _tls_methods:
+            ctx = SSL.Context(method=_tls_methods[os.getenv('TLS_VERSION')])
+        else:
+            # Despite its name, SSLv23_METHOD is the alias for PROTOCOL_TLS which allows for
+            # negotiation between different TLS versions and chooses the highest supported server
+            # protocol. See https://docs.python.org/3/library/ssl.html#ssl.PROTOCOL_SSLv23
+            # specifically PROTOCOL_SSLv23 and PROTOCOL_TLS
+            ctx = SSL.Context(method=SSL.SSLv23_METHOD)
+            # Using SSLv23 by default enables all types of negotiation and we want to limit
+            # negotiation to only 1.2 and 1.3 versions
+            ctx.set_options(SSL.OP_NO_SSLv2)
+            ctx.set_options(SSL.OP_NO_SSLv3)
+            ctx.set_options(SSL.OP_NO_TLSv1)
+            ctx.set_options(SSL.OP_NO_TLSv1_1)
```

Comment:
yes it will use 1.3 but we don't want to block 1.2 because if 1.3 is not available, then there will be a failure. The client provides a list of versions (1.2,1.3) and the server will pick the highest version (1.2 for example)

Reply:
Again I'm new to that stuff so maybe I'm wrong. An attacker will want to downgrade protocol security. that is why the server should allow small number of protocols options as possible. 
This is up to us to decide what should be the default (1.3 only or 1.3 and 1.2) I guess we should choose the nginx tls default version

Reply:
talk on the phone, keeping 1.2 1.3